client/components/domains/use-my-domain/utilities/get-option-info.js

```diff
@@ -35,7 +36,7 @@ export const getDomainTransferrability = ( domainInboundTransferStatusInfo ) =>
 			...optionInfo.transferNotSupported,
 			topText:
 				getTransferRestrictionMessage( domainInboundTransferStatusInfo ) ??
-				optionInfo.transferNotSupported.topText,
+				getUnsupportedDomainTransferMessage( domain ),
```

Comment:
The `if ( ! result.transferrable ) {` condition does not necessarily mean that a domain has a not supported extension, so I think this isn't the best place to change the message - see my other comments

Reply:
Just to add a bit on to what Leo is proposing, I think you could also remove the condition that checks for `! isDomainTransferrable` and add in a call to `getAvailabilityNotice()` in the `default` case of the switch statement.  Then just set the default transfer content to something like:

```
transferContent = {
    ...optionInfo.transferNotSupported,
    topText: availabilityNotice.message,
};
```

(where `availabilityNotice` is the data you get back from the call to `getAvailabilityNotice()`)

This would also allow us to take advantage of all of the pre-existing availability notice data  so we don't have to re-create them.

Reply:
Thanks @delputnam 

I am calling `getAvailabilityNotice()`  like this:

```
import { getAvailabilityNotice } from 'calypso/lib/domains/registration/availability-messages';
const availabilityNotice = getAvailabilityNotice( domain, availability, null );
```

And finally setting the default condition like this:

```
default:
			transferContent = {
				...optionInfo.transferNotSupported,
				topText: availabilityNotice.message,
			};
```

But upon testing I get:

![](https://d.pr/i/B5xRDZ+)
[Direct Link](https://d.pr/i/B5xRDZ+)

The console warning which I understand:

```
Warning: Failed prop type: The prop `availability` is marked as required in `DomainTransferOrConnect`, but its value is `null`.
```

I can see that the propType `availability`  has been set for `DomainTransferOrConnect` as required, is there something I am doing incorrectly that is causing this warning message?